Documentation/Usage.md

```diff
@@ -115,204 +115,257 @@ url.
+```
 
-```swift
-import Clibgit
+Create a module in the `Sources/` directory called `cairo/`, and create two files, `cairo.h` and `module.modulemap`:
 
-let options = git_repository_init_options()
-print(options)
+```bash
+cd Sources 
+mkdir cairo 
+cd cairo 
+touch cairo.h 
+touch module.modulemap
 ```
 
-To `import Clibgit`, the package manager requires that the libgit2 library has
-been installed by a system packager (eg. `apt`, `brew`, `yum`, etc.).  The
-following files from the libgit2 system-package are of interest:
+```
+.
+├── Package.swift
+├── README.md
+└── Sources
+    ├── cairo
+    │   ├── cairo.h
+    │   └── module.modulemap
+    └── example
+        └── main.swift
+```
 
-    /usr/local/lib/libgit2.dylib      # .so on Linux
-    /usr/local/include/git2.h
+The `module.modulemap` file tells Swift how to import and link the system library. Add the following lines to it:
 
-Swift packages that provide module maps for system libraries are handled
-differently from regular Swift packages.
+```
+module cairo {
+    umbrella header "cairo.h"
+    link "cairo"
+}
+```
 
-Note that the system library may be located elsewhere on your system, such as
-`/usr/` rather than `/usr/local/`.
+The `module cairo` declaration specifies the name of the module as Swift sees it. In this example, we will be able to import the library from Swift code with `import cairo`.
+The `umbrella header "cairo.h"` line specifies the path to a C header file to include, in this case, the `cairo.h` file we created.
+The `link "cairo"` line specifies the linker flag used to link the system library, roughly equivalent to `-lcairo` in C.
 
-Create a directory called `Clibgit` next to the `example` directory and
-initialize it as a package that builds a system module:
+In the `cairo.h` file, add the following line:
 
-    example$ cd ..
-    $ mkdir Clibgit
-    $ cd Clibgit
-    Clibgit$ swift package init --type system-module
+```c
+#include <cairo.h>
+```
 
-This creates `Package.swift` and `module.modulemap` files in the directory.
-Edit `Package.swift` and add `pkgConfig` parameter:
+This file functions just like a normal C header, where the `<>` brackets tell `clang` to look for the `cairo.h` header installed by your system in the usual locations. While it’s possible to reference the Cairo header directly in the `module.modulemap` file, including it through a local shim prevents you from having to specify an exact path to it. Many popular C libraries also allow (and even require) you to perform some customization at the inclusion site, so the local header file gives you a place to do this.
 
-```swift
-import PackageDescription
+Next, we have to tell the package manager about the system library module. Go into your `Package.swift` and add a `systemLibrary` target. Don’t forget to specify it as a dependency of your Swift `example` module:
 
+```swift
 let package = Package(
-    name: "Clibgit",
-    pkgConfig: "libgit2"
+    name: "example",
+    dependencies: [], 
```

Comment:
This can be omitted as we're not using any package dependency in this example.

Reply:
done